cmd/sonobuoy/app/pluginList.go

```diff
@@ -108,16 +125,42 @@ func (p *pluginList) loadPluginsDir(dirpath string) error {
 	return nil
 }
 
-// loadSinglePlugin loads a single plugin located at the given path.
-func (p *pluginList) loadSinglePlugin(filepath string) error {
-	b, err := ioutil.ReadFile(filepath)
+// loadSinglePluginFromURL loads a single plugin located at the given path.
+func (p *pluginList) loadSinglePluginFromURL(url string) error {
+	c := http.Client{
+		Timeout: 10 * time.Second,
+	}
+	resp, err := c.Get(url)
+	if err != nil {
+		return errors.Wrapf(err, "unable to GET URL %q", url)
+	}
+	if resp.StatusCode > 399 {
+		return fmt.Errorf("unexpected HTTP response code %v", resp.StatusCode)
+	}
+
+	return p.loadSinglePlugin(resp.Body)
+}
+
+// loadSinglePluginFromFile loads a single plugin located at the given path.
+func (p *pluginList) loadSinglePluginFromFile(filepath string) error {
+	f, err := os.Open(filepath)
 	if err != nil {
 		return errors.Wrapf(err, "unable to read file %q", filepath)
 	}
+	return p.loadSinglePlugin(f)
+}
+
+// loadSinglePlugin reads the data from the reader and loads the plugin.
+func (p *pluginList) loadSinglePlugin(r io.ReadCloser) error {
+	defer r.Close()
+	b, err := ioutil.ReadAll(r)
+	if err != nil {
+		return errors.Wrap(err, "failed to read data for plugin")
+	}
 
 	newPlugin, err := loadManifest(b)
 	if err != nil {
-		return errors.Wrapf(err, "failed to load plugin file %q", filepath)
+		return errors.Wrap(err, "failed to load plugin")
```

Comment:
Minor usability nit: If the user specifies multiple plugins and one fails to decode, it seems like it won't be as easy to determine which one had the issue. Instead, could you return the unwrapped error here and wrap where this function is called in `loadSinglePluginFromURL` and `loadSinglePluginFromFile`? That way both of those functions could include the location information of the plugin in the error message.

Reply:
Instead of just returning the value of `loadSinglePlugin` I wrapped that value with the context necessary to describe the URL or file. 👍